docs/guide/CSS.md

```diff
@@ -0,0 +1,395 @@
+
+### ID selector
+
+Every Widget can have a single `id` attribute, which is set via the constructor. The ID should be unique to it's container.
+
+Here's an example of a widget with an ID:
+
+```python
+yield Button(id="next")
+```
+
+You can match an ID with a selector starting with a hash (`#`). Here is how you might draw a red outline around the above button:
+
+```sass
+#next {
+  outline: red;
+}
+```
+
+A Widget's `id` attribute can not be changed after the Widget has been constructed. 
+
+### Class-name selector
+
+Every widget can have a number of class names applied. The term "class" here is borrowed from web CSS, and has a different meaning to a Python class. You can think of a CSS class as a tag of sorts. Widgets with the same tag will share styles.
+
+CSS classes are set via the widgets `classes` parameter in the constructor. Here's an example:
+
+```python
+yield Button(classes="success")
+```
+
+This button will have a single class called `"success"` which we could target via CSS to make the button a particular color.
+
+You may also set multiple classes separated by spaces. For instance, here is a button with both an `error` class and a `disabled` class:
+
+```python
+yield Button(classes="error disabled")
+```
+
+To match a Widget with a given class in CSS you can precede the class name with a dot (`.`). Here's a rule with a class selector to match the `"success"` class name:
+
+```sass
+.success {
+  background: green;
+  color: white;
+}
+```
+
+!!! note
+
+    You can apply a class name to any class, which means that widgets of different types could share classes.
+
+Class name selectors may be _chained_ together by appending another full stop and class name. The selector will match a widget that has _all_ of the class names set. For instance, the following sets a red background on widgets that have both `error` _and_ `disabled` class names.
+
+```sass
+.error.disabled {
+  background: darkred;
+}
+```
+
+Unlike the `id` attribute a Widget's classes can be changed after the Widget was created. Adding and removing CSS classes is the recommended way of changing the display while your app is running. There are a few methods you can use to manage CSS classes.
+
+- [add_class()][textual.dom.DOMNode.add_class] Adds one or more classes to a widget.
+- [remove_class()][textual.dom.DOMNode.remove_class] Removes class name(s) from a widget.
+- [toggle_class()][textual.dom.DOMNode.toggle_class] Removes a class name if it is present, or adds the name if its not already present.
+- [has_class()][textual.dom.DOMNode.has_class] Checks if a class(es) is set on a widget.
+- [classes][textual.dom.DOMNode.classes] Is a frozen set of the class(es) set on a widget.
+
+
+### Universal selector
+
+The _universal_ selectors is denoted by an asterisk and will match _all_ widgets.
+
+For example, the following will draw a red outline around all widgets:
+
+```sass
+* {
+  outline: solid red;
+}
+```
+
+### Pseudo classes
```

Comment:
Pseudo class or pseudo selector? Both are used below, seemingly interchangeably. Might be worth picking one, or clarifying they mean the same thing.

Reply:
They don't quite mean the same thing. Pseudo classes are the attributes on the widget, pseudo selectors are the CSS syntax.